plugins/chat/assets/javascripts/discourse/components/chat-channel.js

```diff
@@ -213,6 +213,12 @@ export default class ChatLivePane extends Component {
               thread,
               { replace: true }
             );
+
+            storedThread.tracking.unreadCount =
```

Comment:
Since we are replacing the thread here and below we need to make sure the tracking state is correctly populated too, otherwise we can load a thread in the list and then the channel and the channel thread clobbers the list one.